### PR TITLE
Fix for cjs issues with camelids + regression tests + better types

### PR DESCRIPTION
**Pull request by carsonfarmer**

### CODE REVIEW

**Review by joewagner — APPROVED** [review]
All looks great! only thing I can see is I think there might be a typo in a comment

File: test/registry.test.ts [hunk 1]
```diff
@@ -36,6 +37,17 @@ describe("registry", function () {
       strictEqual(receipt.chainId, 31337);
     });
 
+    test("regression: when a tx comes back without a chainId", async function () {
+      const tx = await reg.createTable({
+        chainId: 31337,
+        statement: "create table test_no_chainid_31337 (id int, name text)",
+      });
+      tx.chainId = 0; // Wipe out chainId information, which can happen
+      // with MetaMask if no provided is connected
```
Comment (joewagner) [inline test/registry.test.ts:46]: Should this be provider?

**Review by asutula — APPROVED** [review]

**Comment by joewagner** [comment]
I think the tests on this pr are failing because of the wasm-sqlparser cjs build.
https://github.com/tablelandnetwork/wasm-sqlparser/pull/97 should fix the problem.

**Comment by joewagner** [comment]
@carsonfarmer @asutula  All the tests are passing, do either of you have any manual tests you want to run to check this?
If all is good, should we do another pre-release?

**Comment by asutula** [comment]
I'll try to integrate here locally just to verify things work.

**Comment by asutula** [comment]
Looks like it works! Merge away.

**Pull request merged**